source/s3_checksum_stream.c

```
@@ -11,18 +11,24 @@ struct aws_checksum_stream {
     struct aws_input_stream *old_stream;
     struct aws_s3_checksum *checksum;
     struct aws_byte_buf checksum_result;
-    struct aws_byte_buf *encoded_checksum_result;
+    /* base64 encoded checksum of the stream, updated on destruction of sream */
+    struct aws_byte_buf *encoded_checksum_output;
     bool did_seek;
 };
 
 static int s_aws_input_checksum_stream_seek(
     struct aws_input_stream *stream,
     int64_t offset,
     enum aws_stream_seek_basis basis) {
-
-    struct aws_checksum_stream *impl = stream->impl;
-    impl->did_seek = true;
-    return aws_input_stream_seek(impl->old_stream, offset, basis);
+    (void)stream;
+    (void)offset;
+    (void)basis;
+    AWS_LOGF_ERROR(
+        AWS_LS_S3_CLIENT,
+        "Cannot seek on checksum stream, as it will cause the checksum output to mismatch the checksum of the stream "
+        "contents");
+    AWS_ASSERT(false);
+    return AWS_OP_ERR;
```

you need to raise an error. found or define an error to raise here.